src/lib/masq.js

```diff
@@ -606,11 +643,54 @@ class Masq {
     peer.send(JSON.stringify(encryptedMsg))
   }
 
-  _createDBAndSyncApp (dbName) {
-    return new Promise((resolve, reject) => {
-      const db = openOrCreateDB(dbName)
+  _watchAndAuthorizeApps () {
+    const syncApps = async () => {
+      const allDevices = await this.getDevices()
+      const otherDevices = allDevices.filter(d => {
+        return d.localKey !== this.profileDB.local.key.toString('hex')
+      })
+
+      for (let otherDevice of otherDevices) {
+        for (let app of otherDevice.apps) {
+          if (!this.appsDBs[app.discoveryKey]) {
+            // Add app
+            await this._createDBAndSyncApp(app.id + '-copy', app.key)
```

Comment:
Because in the context of the tests, replication will occur in the same browser and tab. Hence, we need to replicate between two different IndexedDB databases.